integration-cli/docker_api_containers_test.go

```diff
@@ -320,6 +320,49 @@ func TestGetContainerStats(t *testing.T) {
 	logDone("container REST API - check GET containers/stats")
 }
 
+func TestGetStoppedContainerStats(t *testing.T) {
+	defer deleteAllContainers()
+	var (
+		name   = "statscontainer"
+		runCmd = exec.Command(dockerBinary, "create", "--name", name, "busybox", "top")
+	)
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatalf("Error on container creation: %v, output: %q", err, out)
+	}
+	type b struct {
+		body []byte
+		err  error
+	}
+	bc := make(chan b, 1)
+	go func() {
+		body, err := sockRequest("GET", "/containers/"+name+"/stats", nil)
+		bc <- b{body, err}
+	}()
+
+	// allow some time to stream the stats from the container
+	time.Sleep(4 * time.Second)
+	if _, err := runCommand(exec.Command(dockerBinary, "rm", "-f", name)); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This section is not needed as for your test there will not be anything returned from `/stats` endpoint; and your test is to validate that it doesn't panic at this point (without your fix)!  This adds time to the tests that isn't really valuable, so would be better to remove this whole block and add rely on the `deleteAllContainers()` you already have at the beginning of your test function.